File: src/main/java/ejerciciosJava/Ejercicio10.java
package ejerciciosJava;

import java.util.Scanner;

public class Ejercicio10 {
    public static void main() {
        Scanner scanner = new Scanner(System.in);
        Boolean celsius = null;

        while (celsius == null) {
            System.out.println("Introduce a qué temperatura quieres convertir (C o F)");
            String userA = scanner.nextLine().toLowerCase();
            if (userA.equals("c")) {
                celsius = true;
            } else if (userA.equals("f")) {
                celsius = false;
            } else {
                System.out.println("Vuelve a intentarlo");
            }
        }

        int numeroConvertir = Utils.pedirEntero("Introduce una temperatura: ");
        double numeroConvertido = celsius ? (numeroConvertir - 32) * 5.0 / 9 : numeroConvertir * 9.0 / 5 + 32;
        String tString = celsius ? "Celsius" : "Fahrenheit";
        System.out.println("Temperatura en " + tString + " = " + numeroConvertido);
    }
}

File: src/main/java/ejerciciosKotlin/Ejercicio10.kt
package ejerciciosKotlin

fun ejercicio10(){
    var celsius:Boolean? = null
    while (celsius == null){
        println("Introduce a qué temperatura quieres convertir (C o F)")
        val userA = readln().lowercase()
        when(userA){
            "c" -> celsius = true
            "f" -> celsius = false
            else -> println("Vuelve a intentarlo")
        }
    }

    val numeroConvertir = pedirEntero("Introduce una temperatura: ")
    val numeroConvertido = if (celsius) (numeroConvertir - 32) * 5/9 else numeroConvertir * 9/5 + 32
    val tString = if (celsius) "Celsius" else "Fahrenheit"
    println("Temperatura en $tString = $numeroConvertido")

}